### PR TITLE
Support link option

### PR DESCRIPTION
**Pull request by Konboi**
Add `--link` option to the `launchable record` command and `launchable record session` command. Also, `launchable subset` command and `launchable record tests` command support a link option to pass links value to the `launchable record session` command internally too.

e.g)
```sh
# record build
$ launchable record build --name test --link build-url=https://example.launchable.com/job/1 

# record session
$ launchable record build --name test --link commit=https://github.com/owner/repo/commits/123abc456
```

Also, if these values (ref: https://github.com/launchableinc/cli/pull/529/files#diff-280a9bc4f9d9c38c29d666a70adf8c4e216a1abb22f548b3a011f9ac00b6bf10R4-R53) are set in the CI environment, the cli sets custom link data automatically.

### CODE REVIEW

File: launchable/commands/record/build.py [hunk 1]
```diff
@@ -63,9 +66,17 @@
     default=[],
     cls=KeyValueType,
 )
+@click.option(
+    '--link',
+    'links',
+    help="Set external link of title and url",
```
Comment (shibuiwilliam) [inline launchable/commands/record/build.py:72]: [question]
Is this option requires both `title and url`, like dict format?

File: launchable/commands/record/build.py [hunk 2]
```diff
@@ -172,9 +176,19 @@ def build(ctx: click.core.Context, build_name: str, source: List[str], max_days:
...
+            "commitHashes": commitHashes,
         }
 
+        _links = capture_link(os.environ)
+        if len(links) != 0:
+            for link in normalize_key_value_types(links):
+                _links.append({
+                    "title": link[0],
+                    "url": link[1],
+                    "kind": "CUSTOM_LINK"
```
Comment (shibuiwilliam) [inline launchable/commands/record/build.py:188]: [nits]
Probably better to use `LinkKind.CUSTOM_LINK`.
https://github.com/launchableinc/cli/pull/529/files#diff-280a9bc4f9d9c38c29d666a70adf8c4e216a1abb22f548b3a011f9ac00b6bf10R18

File: launchable/commands/record/session.py [hunk 3]
```diff
@@ -73,16 +74,23 @@ def session(
...
-    if link:
-        payload["link"] = link
+
+    _links = capture_link(os.environ)
+    if len(links) != 0:
+        for link in normalize_key_value_types(links):
+            _links.append({
+                "title": link[0],
+                "url": link[1],
+                "kind": "CUSTOM_LINK"
```
Comment (shibuiwilliam) [inline launchable/commands/record/session.py:91]: [nits]
Probably better to use `LinkKind.CUSTOM_LINK`.
https://github.com/launchableinc/cli/pull/529/files#diff-280a9bc4f9d9c38c29d666a70adf8c4e216a1abb22f548b3a011f9ac00b6bf10R18

File: launchable/utils/link.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+JENKINS_URL_KEY = 'JENKINS_URL'
+JENKINS_BUILD_URL_KEY = 'BUILD_URL'
+GITHUB_ACTIONS_KEY = 'GITHUB_ACTIONS'
+GITHUB_ACTIONS_SERVER_URL_KEY = 'GITHUB_SERVER_URL'
+GITHUB_ACTIONS_REPOSITORY_KEY = 'GITHUB_REPOSITORY'
+GITHUB_ACTIONS_RUN_ID_KEY = 'GITHUB_RUN_ID'
+GITHUB_PULL_REQUEST_URL_KEY = "GITHUB_PULL_REQUEST_URL"
+CIRCLECI_KEY = 'CIRCLECI'
+CIRCLECI_BUILD_URL_KEY = 'CIRCLE_BUILD_URL'
```
Comment (shibuiwilliam) [inline launchable/utils/link.py:12]: [question]
CIRCLECI_BUILD_URL_KEY = 'CIRCLECI_BUILD_URL' ?

File: launchable/utils/link.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            {"kind": LinkKind.JENKINS.value, "url": env.get(JENKINS_BUILD_URL_KEY, ""), "title": ""}
+        )
+    if env.get(GITHUB_ACTIONS_KEY):
+        links.append({
+            "kind": LinkKind.GITHUB_ACTIONS.value,
+            "url": "{}/{}/actions/runs/{}".format(
+                env.get(GITHUB_ACTIONS_SERVER_URL_KEY),
+                env.get(GITHUB_ACTIONS_REPOSITORY_KEY),
+                env.get(GITHUB_ACTIONS_RUN_ID_KEY),
+            ),
```
Comment (shibuiwilliam) [inline launchable/utils/link.py:39]: [question]
What if there is no GITHUB_ACTIONS_SERVER_URL_KEY set?
I feel it requires env exists validation.

File: launchable/utils/link.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class LinkKind(Enum):
+
+    LINK_KIND_UNSPECIFIED = "LINK_KIND_UNSPECIFIED"
+    CUSTOM_LINK = "CUSTOM_LINK"
+    JENKINS = "JENKINS"
+    GITHUB_ACTIONS = "GITHUB_ACTIONS"
+    GITHUB_PULL_REQUEST = "GITHUB_PULL_REQUEST"
+    CIRCLECI = "CIRCLECI"
```
Comment (shibuiwilliam) [inline launchable/utils/link.py:22]: [fyi]
In Python Enum, it is possible to string of enum key by `.name`.

```python
class LinkKind(Enum):
    LINK_KIND_UNSPECIFIED

print(LinkKind.LINK_KIND_UNSPECIFIED.name)
# LINK_KIND_UNSPECIFIED
```

File: launchable/commands/record/build.py (hunk 1 above)
Comment (Konboi) [reply launchable/commands/record/build.py:72]: Yes, like this ` --link build-url=https://example.launchable.com/job/1`

File: launchable/utils/link.py (hunk 4 above)
Comment (Konboi) [reply launchable/utils/link.py:12]: `CIRCLE_BUILD_URL` is correct 
image

ref: https://circleci.com/docs/variables/

File: launchable/utils/link.py (hunk 5 above)
Comment (Konboi) [reply launchable/utils/link.py:39]: These values will be set automatically by GitHub Actions ref: https://docs.github.com/en/actions/learn-github-actions/variables

But, you mean that we should validate these values, right?

File: launchable/utils/link.py (hunk 6 above)
Comment (Konboi) [reply launchable/utils/link.py:22]: Is it true?

I tried to make sample codes then faced `NameError: name 'LINK_KIND_UNSPECIFIED' is not defined`

```python
from enum import Enum

class LinkKind(Enum):
    LINK_KIND_UNSPECIFIED

print(LinkKind.LINK_KIND_UNSPECIFIED.name)
```
Do we need to use `auto()?

Comment (shibuiwilliam) [reply launchable/utils/link.py:22]: Sorry, you are right. You need auto().

```python
from enum import Enum, auto

class LinkKind(Enum):
    LINK_KIND_UNSPECIFIED = auto()

print(LinkKind.LINK_KIND_UNSPECIFIED.name)
# LINK_KIND_UNSPECIFIED
```

File: launchable/utils/link.py (hunk 4 above)
Comment (shibuiwilliam) [reply launchable/utils/link.py:12]: ok

File: launchable/commands/record/build.py (hunk 1 above)
Comment (shibuiwilliam) [reply launchable/commands/record/build.py:72]: ok

File: launchable/utils/link.py (hunk 5 above)
Comment (shibuiwilliam) [reply launchable/utils/link.py:39]: Yes, I meant to say I feel it is necessary to validate these values exist in env var, but if they always exist in GitHub Actions, that validation is not needed.

Comment (Konboi) [reply launchable/utils/link.py:39]: OK, so I won't add validations

**Review by shibuiwilliam — APPROVED** [review]
CI is failing. Please fix.

File: launchable/utils/link.py (hunk 6 above)
Comment (shibuiwilliam) [reply launchable/utils/link.py]: How about this for Python3.5?

```python
from enum import Enum

class LinkKind(Enum):
    LINK_KIND_UNSPECIFIED = 0

print(LinkKind.LINK_KIND_UNSPECIFIED.name)
# LINK_KIND_UNSPECIFIED
```

Comment (Konboi) [reply launchable/utils/link.py]: try to test https://github.com/launchableinc/cli/pull/529/commits/e606695dea187045a734b1b6af2149c0ba61432a

**Review by shibuiwilliam — APPROVED** [review]

**Comment by Konboi** [comment]
@draftcode Could you review this PR?

**Review by draftcode — APPROVED** [review]

**Pull request merged**